src/components/map-overlays/map-overlays.component.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect, useState } from "react";
+import { Polygon, Overlay } from "react-native-maps";
+import { Buildings } from "../../constants/buildings.data";
+import { BuildingId, ZoomLevel, IndoorInformation } from "../../types/main";
+import { View, StyleSheet } from "react-native";
+import { CONCORDIA_RED, BUILDING_UNTAPPED } from "../../constants/style";
+import { getAllCampuses } from "../../constants/campus.data";
+import { floorOverlays } from "../../constants/floors.data";
+import { getAllPOI } from "../../constants/poi.data";
+import CustomMarker from "../custom-marker/custom-marker.component";
+
+interface IProps {
+  onBuildingTap: (id: BuildingId) => void;
+  tappedBuilding: BuildingId;
+  zoomLevel: ZoomLevel;
+  indoorInformation: IndoorInformation;
+}
+
+/**
+ * Wrapper Component for Polygons and Markers which overlay campus buildings.
+ */
+const MapOverlays = ({
+  onBuildingTap,
+  tappedBuilding,
+  zoomLevel,
+  indoorInformation
+}: IProps) => {
+  /**
+   * Fill color for the Polygons
+   */
+  const [fillColor, setFillColor] = useState<string>(null);
+  const [tappedColor, setTappedColor] = useState<string>(null);
+  useEffect(() => {
+    setTimeout(() => {
+      setFillColor(BUILDING_UNTAPPED);
+      setTappedColor(CONCORDIA_RED);
+    }, 50);
+  }, []);
+
+  return (
+    <>
```

Comment:
Need comments to know what overlays these are

Reply:
Done! Good point